app/src/androidTest/java/org/akvo/flow/activity/form/submittedformsview/FormActivityReadOnlyTest.java

```diff
@@ -0,0 +1,483 @@
+        String questionValue = getResponseValue(question);
+        if (!TextUtils.isEmpty(questionValue)) {
+            String[] items = questionValue.split("\\|");
+            ViewInteraction barcodeInput = onView(
+                    allOf(withId(R.id.barcode_responses_recycler_view),
+                            withQuestionViewParent(question,
+                                    BarcodeQuestionViewReadOnly.class))).perform(scrollTo());
+            barcodeInput.check(matches(isDisplayed()));
+            for (int i = 0; i < items.length; i++) {
+                barcodeInput.perform(scrollToPosition(i));
+                barcodeInput.check(matches(hasDescendant(withText(items[i]))));
+            }
+        }
+    }
+
+    private void verifyDateQuestionView(Question question) {
+        verifyDateInput(question);
+        verifyDateButton(question);
+    }
+
+    private void verifyDateInput(Question question) {
+        String questionValue = getResponseValue(question);
+        ViewInteraction dateInput = getDateEditText(question);
+        dateInput.check(matches(isDisplayed()));
+        DateFormat userDisplayedDateFormat = SimpleDateFormat.getDateInstance();
+        userDisplayedDateFormat.setTimeZone(TimeZone.getDefault());
+        if (questionValue != null) {
+            dateInput.check(matches(withText(
+                    userDisplayedDateFormat.format(new Date(Long.parseLong(questionValue))))));
+        } else {
+            dateInput.check(matches(withText("")));
+        }
+    }
+
+    private void verifyDateButton(Question question) {
+        ViewInteraction dateButton = getDateButton(question);
+        dateButton.check(matches(isDisplayed()));
+        dateButton.check(matches(not(isEnabled())));
+        dateButton.check(matches(withText(R.string.pickdate)));
+    }
+
+    private void verifyPhotoQuestionView(Question question) {
+        verifyMediaButton(question, R.string.takephoto);
+        verifyMediaContent(question);
+    }
+
+    private void verifyMediaButton(Question question, int textResId) {
+        ViewInteraction mediaButton = getMediaButton(question);
+        mediaButton.check(matches(withText(textResId)));
+        mediaButton.check(matches(allOf(isDisplayed(), not(isEnabled()))));
+    }
+
+    private void verifyVideoQuestionView(Question question) {
+        verifyMediaButton(question, R.string.takevideo);
+        verifyMediaContent(question);
+    }
+
+    private void verifyMediaContent(Question question) {
+        String questionValue = getResponseValue(question);
+        if (!TextUtils.isEmpty(questionValue)) {
+            ViewInteraction downloadButton = onView(
+                    allOf(withId(R.id.media_download),
+                            withQuestionViewParent(question, MediaQuestionView.class)))
+                    .perform(scrollTo());
+            downloadButton.check(matches(isDisplayed()));
+            downloadButton.perform(click());
```

Comment:
It should but it would be more appropriate in a detailed test on that question type in particular. I have removed it and just added a check to make sure the button is enabled.